book.md

```diff
@@ -510,6 +510,154 @@ can also implement automatic bounds checking for arrays where the array bound
+The example execution below tricks the software into thinking the user is logged in, by
+taking advantage of the UaF to change the boolean:
+
+```
+> auth admin                                   
+[ auth = 0x78322e1000, service = 0x0 ]       
+> reset                                        
+[ auth = 0x78322e1000, service = 0x0 ]       
+> service aaaaaaaaa0aaaaaaaaa0aaaaaaaaa0121    
+[ auth = 0x78322e1000, service = 0x78322e1000 ]
+> login                                        
+You are already logged in!
+```
+
+In this example, four commands are issued:
+1) 'auth admin'  will allocate the auth\_t structure for the first time. At
+this point, the user has a name, but is not authorized (logged\_in is false).
+2) 'reset'  will free the memory (but pointer to auth is not set to null).
+service (...) will allocate a new string, potentially at the same address
+where the auth\_t structure was previously allocated. 
+3) 'service aaaaaaaaa0aaaaaaaaa0aaaaaaaaa0121' will end up setting the memory
+previously pointing to an auth\_t.
+4) 'login' will use the dangling auth\_t pointer. If this memory has been
+reallocated to the attacker controlled string, it will appear as if the field
+"name" is set to aaaaaaaaa0aaaaaaaaa0aaaaaaaaa012 and the boolean
+logged\_in to true.
+
+:::
+
+Detecting UaFs is usually not an easy task, as it depends not only on
+user inputs, but sometimes also on the execution flow. It can get even more
+complicated in multi-threaded environments. Some UaF detectors intercept calls
+to delete / free to inject a known value to the variables and then run the
+software looking for crashes that include that value. Another useful tool for
+detecting UaF is Arm's MTE (Memory Tagging Extension), discussed in
+section @sec:preventing-and-detecting-memory-errors.
+Fuzzing (generating random inputs) can also be useful, potentially used at
+the same time as other tools.
+
+Preventing UaF from happening may involve multiple approaches, depending on the
+context. From simple code changes, such as initializing allocated variables, to
+more elaborate changes, such as changing how the memory allocator works to
+avoid reusing specific memory locations. On top of that, decreasing relevance
+of UaF for attackers can be another interesting perspective (meaning, even if
+a UaF is present, decreasing the likelihood of it being exploitable). For
+example, MTE sync mode can force an application crash as soon as the UaF occurs,
+while Pointer Authentication (PAC) can be used to sign pointers so that even if
+they get poisoned, they cannot be used (more details in
+section @sec:pointer-authentication.
+
+Those are not exaustive lists of tools to detect, mitigate and prevent UaFs.
+One could also include different algorithms to reduce the probability of exploiting
+use-after-free in allocators as a mitigation strategy, garbage collectors or
+reference-count based memory allocators. But this section's goal was to give
```

Comment:
Shouldn't this sentence go into the above paragraph that has the list of method to reduce the potential impact of a UaF?